### PR TITLE
Upgrade nbconvert req; remove jinja2 dependency

### LINKED ISSUE #3403: Unpin jinja2
In #3402, we pinned jinja2 to resolve document building issues. This issue tracks unpinning jinja2 on the resolution of https://github.com/sphinx-doc/sphinx/issues/10291.

### PR DESCRIPTION
**Pull request by rwedge**
Fixes #3403

### CODE REVIEW

File: docs-requirements.txt [hunk 1]
```diff
@@ -1,11 +1,10 @@
-jinja2 == 3.0.3
```
Comment (jeremyliweishih) [inline docs-requirements.txt:1]: might need to pin this still until jinja resolves, still getting: `ImportError: cannot import name 'environmentfilter' from 'jinja2' (/home/docs/checkouts/readthedocs.org/user_builds/feature-labs-inc-evalml/envs/3411/lib/python3.8/site-packages/jinja2/__init__.py)`

Comment (rwedge) [reply docs-requirements.txt:1]: looks like an old version of sphinx

**Comment by rwedge** [comment]
Has this error come up before?

```python-traceback
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/feature-labs-inc-evalml/envs/3411/lib/python3.8/site-packages/ipykernel/kernelbase.py", line 1200, in _at_shutdown
    await self._progressively_terminate_all_children()
GeneratorExit

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/feature-labs-inc-evalml/envs/3411/lib/python3.8/site-packages/ipykernel/kernelbase.py", line 297, in process_control
    await result
  File "/home/docs/checkouts/readthedocs.org/user_builds/feature-labs-inc-evalml/envs/3411/lib/python3.8/site-packages/ipykernel/kernelbase.py", line 836, in shutdown_request
    await self._at_shutdown()
  File "/home/docs/checkouts/readthedocs.org/user_builds/feature-labs-inc-evalml/envs/3411/lib/python3.8/site-packages/ipykernel/kernelbase.py", line 1206, in _at_shutdown
    self.session.send(
  File "/home/docs/checkouts/readthedocs.org/user_builds/feature-labs-inc-evalml/envs/3411/lib/python3.8/site-packages/jupyter_client/session.py", line 853, in send
    stream.send_multipart(to_send, copy=copy)
  File "/home/docs/checkouts/readthedocs.org/user_builds/feature-labs-inc-evalml/envs/3411/lib/python3.8/site-packages/ipykernel/iostream.py", line 285, in send_multipart
    return self.io_thread.send_multipart(*args, **kwargs)
  File "/home/docs/checkouts/readthedocs.org/user_builds/feature-labs-inc-evalml/envs/3411/lib/python3.8/site-packages/ipykernel/iostream.py", line 225, in send_multipart
    self.schedule(lambda : self._really_send(*args, **kwargs))
  File "/home/docs/checkouts/readthedocs.org/user_builds/feature-labs-inc-evalml/envs/3411/lib/python3.8/site-packages/ipykernel/iostream.py", line 218, in schedule
    f()
  File "/home/docs/checkouts/readthedocs.org/user_builds/feature-labs-inc-evalml/envs/3411/lib/python3.8/site-packages/ipykernel/iostream.py", line 225, in <lambda>
    self.schedule(lambda : self._really_send(*args, **kwargs))
  File "/home/docs/checkouts/readthedocs.org/user_builds/feature-labs-inc-evalml/envs/3411/lib/python3.8/site-packages/ipykernel/iostream.py", line 233, in _really_send
    self.socket.send_multipart(msg, *args, **kwargs)
AttributeError: 'NoneType' object has no attribute 'send_multipart'
```

File: docs/source/conf.py [hunk 2]
```diff
@@ -284,7 +284,7 @@ def setup(app):
         shutil.copy("disable-warnings.py", "/home/docs/.ipython/profile_default/startup/")
         shutil.copy("set-headers.py", "/home/docs/.ipython/profile_default/startup")
     app.add_domain(PatchedPythonDomain, override=True)
-    app.add_javascript(
+    app.add_js_file(
        "https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js"
     )
     app.add_stylesheet("style.css")
```
Comment (jeremyliweishih) [inline docs/source/conf.py:290]: one last warning here from the build:
```
WARNING: while setting up extension conf.py: The app.add_stylesheet() is deprecated. Please use app.add_css_file() instead.
```

**Review by jeremyliweishih — APPROVED** [review]
looks good after the last warning fix!

**Comment by rwedge** [comment]
@chukarsten ok to merge this in?

**Comment by chukarsten** [comment]
Thanks @rwedge !!

**Pull request merged**